build.yaml

```diff
@@ -3093,45 +3097,39 @@ steps:
       BATCH_ID=$(hailctl batch submit --name=test-hailctl-batch-submit --files=foo -o json hail_with_args.py 100 | jq '.id')
       STATUS=$(hailctl batch wait -o json $BATCH_ID)
       STATE=$(echo $STATUS | jq -jr '.state')
-      if [ "$STATE" == "success" ]; then
-          exit 0;
-      else
+      if [ "$STATE" != "success" ]; then
           echo $STATUS;
           exit 1;
       fi
 
-      cat >file.sh <<EOF
+      cat >file.sh <<'EOF'
```

Comment:
Is this a typo?

Reply:
No, it prevents interpolation of variables inside the heredoc. From the bash man page (key part is "If any characters in word are quoted, [...] the lines in the here-document are not expanded"):
> Here Documents
>     This type of redirection instructs the shell to read input from the current source until a line containing only word (with no trailing
>     blanks) is seen.  All of the lines read up to that point are then used as the standard input for a command.
>
>     The format of here-documents is:
>
>            <<[-]word
>                    here-document
>            delimiter
>
>     No parameter expansion, command substitution, arithmetic expansion, or pathname expansion is performed on word.  If any characters in word
>     are quoted, the delimiter is the result of quote removal on word, and the lines in the here-document are not expanded.  If word is unquoted,
>     all lines of the here-document are subjected to parameter expansion, command substitution, and arithmetic expansion.  In the latter case,
>     the character sequence \<newline> is ignored, and \ must be used to quote the characters \, $, and `.
>
>     If the redirection operator is <<-, then all leading tab characters are stripped from input lines and the line containing delimiter.  This
>     allows here-documents within shell scripts to be indented in a natural fashion.